cdp/webhook.py

```diff
@@ -0,0 +1,192 @@
+            event_filters (List[WebhookEventTypeFilter]): Filters applied to the events that determine which specific address(es) trigger
+            network_id (str): The network ID of the wallet. Defaults to "base-sepolia".
+
+        Returns:
+            Webhook: The created webhook object.
+
+        """
+        create_webhook_request = CreateWebhookRequest(
+                network_id=network_id,
+                event_type=event_type,
+                event_type_filter=event_type_filter,
+                event_filters=event_filters,
+                notification_uri=notification_uri,
+        )
+
+        model = Cdp.api_clients.webhooks.create_webhook(create_webhook_request)
+        webhook = cls(model)
+
+        return webhook
+
+    @classmethod
+    def list(cls) -> Iterator["Webhook"]:
+        """List webhooks.
+
+        Returns:
+            Iterator[Webhook]: An iterator of webhook objects.
+
+        """
+        while True:
+            page = None
+
+            response: WebhookList = Cdp.api_clients.webhooks.list_webhooks(limit=100, page=page)
+
+            for webhook_model in response.data:
+                yield cls(webhook_model)
+
+            if not response.has_more:
+                break
+
+            page = response.next_page
+
+    @classmethod
+    def delete(cls, webhook_id: str) -> None:
+        """Delete a webhook by its ID.
+
+        Args:
+            webhook_id (str): The ID of the webhook to delete.
+
+        """
+        Cdp.api_clients.webhooks.delete_webhook(webhook_id)
+
+    def update(
+        self,
+        notification_uri: str | None = None,
+        event_type_filter: WebhookEventTypeFilter | None = None
+    ) -> "Webhook":
+        """Update the webhook with a new notification URI, and/or a new list of addresses to monitor.
+
+        Args:
+            notification_uri (str): The new URI for webhook notifications.
+            event_type_filter (WebhookEventTypeFilter): The new eventTypeFilter that contains a new list (replacement) of addresses to monitor for the webhook.
+
+        Returns:
+            Webhook: The updated webhook object.
+
+        """
+        # Fallback to current properties if no new values are provided
+        final_notification_uri = notification_uri or self.notification_uri
+        final_event_type_filter = event_type_filter or self.event_type_filter
+
+        update_webhook_request = UpdateWebhookRequest(
+            event_type_filter=final_event_type_filter,
+            event_filters=self.event_filters,
+            notification_uri=final_notification_uri,
+        )
+
+        # Update the webhook via the API client
+        result = Cdp.api_clients.webhooks.update_webhook(
+            self.id,
+            update_webhook_request,
+        )
+
+        # Update the internal model with the API response
+        self._model = result
+
+        return result
```

Comment:
This should return `Webhook` instead of the underlying model

Reply:
done